iot-hub/Tutorials/Routing/SimulatedDevice/Program.cs

```diff
@@ -128,12 +142,20 @@ private static async Task SendDeviceToCloudMessagesAsync(CancellationToken token
                 message.Properties.Add("level", levelValue);
 
                 // Submit the message to the hub.
-                await s_deviceClient.SendEventAsync(message);
+                try
+                {
+                    await s_deviceClient.SendEventAsync(message, token);
+                }
+                catch (OperationCanceledException) { }
 
                 // Print out the message.
                 Console.WriteLine("{0} > Sent message: {1}", DateTime.UtcNow, telemetryDataString);
 
-                await Task.Delay(1000);
+                try
+                {
+                    await Task.Delay(1000, token);
```

Comment:
Why wait a second?

Reply:
I think the idea is that this is the time between telemetry messages, since this loop just runs until cancelled by the user.